### PR TITLE
Convert spring rabbitmq tests from groovy to java

### PR DESCRIPTION
**Pull request by abhisesh**
Related to https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7195.

Converts spring rabbitmq tests from groovy to java.

### CODE REVIEW

File: instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/java/ContextPropagationTest.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    }
+    if (applicationContext != null) {
+      applicationContext.close();
+    }
+  }
+
+  @Test
+  public void testWithoutHeaders() throws Exception {
+    test(false);
+  }
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/java/ContextPropagationTest.java:89]: Can you use `@ParameterizedTest` instead?

File: instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/network/ClientAttributesExtractorInetSocketAddressTest.java
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/network/ClientAttributesExtractorInetSocketAddressTest.java:1]: @abhisesh can you clean up your branch? Looks like a lot of unrelated changes got in with the merge

File: instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/java/ContextPropagationTest.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
+/*
```
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply instrumentation/spring/spring-rabbit-1.0/javaagent/src/test/java/ContextPropagationTest.java:1]: @abhisesh can you move the test to an `io.opentelemetry.javaagent.instrumentation.spring.rabbit.v1_0` package, as Lauri suggested?

File: instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/network/ClientAttributesExtractorInetSocketAddressTest.java
Comment (abhisesh) [reply instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/network/ClientAttributesExtractorInetSocketAddressTest.java:1]: Sorry missed this one, it's fixed now

**Review by mateuszrzeszutek — APPROVED** [review]
Thanks @abhisesh !
